File: 13_roman_to_integer/pramoda/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_map = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
        }
        num = 0
        last_num = 0
        for i in range(len(s)-1, -1, -1):
            n = roman_map[s[i]]
            if n < last_num:
                num -= n
            else:
                num += n

            last_num = n

        return num
    
s = Solution()
print(s.romanToInt("MCMXCIV"))

File: 13_roman_to_integer/pramoda/solution.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int romanToInt(char* s) {
    //                            0,   1, 2, 3, 4, 5, 6, 7, 8,  9,   10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21
    static const int lut[22] = {100, 500, 0, 0, 0, 0, 1, 0, 0, 50, 1000,  0,  0,  0,  0,  0,  0,  0,  0,  5,  0, 10};

    int length = strlen(s);
    int num = 0;
    int last_num = 0;
    for (int i = length - 1; i >= 0; --i) {
        int n = lut[(s[i] - 'C')];
        num += (n < last_num) ? -n : n;
        last_num = n;
    }
    return num;
}

int main() {
    char s[100] = "MCMXCIV";
    printf("%d\n", romanToInt(s));
    return 0;
	
}
